### PR TITLE
fix(flashpoint): guard against None fields in alert name and media_id concatenation (#7268, #7269)

### LINKED ISSUE #7268: fix(flashpoint): uncaught TypeError when concatenating None field in _generate_incident_name
## Description
The Flashpoint Alerts connector raises an uncaught `TypeError` while generating the incident name for a converted alert, when a field used in a string concatenation is `None` instead of a string.

Observed error:
```
An error occurred while converting alert.id: <alert_id>, error: can only concatenate str (not "NoneType") to str
```

Traceback:
```
File "/opt/opencti-connector-flashpoint/flashpoint_connector/converter_to_stix.py", line 470, in alert_to_incident
    incident_name = self._generate_incident_name(alert)
File "/opt/opencti-connector-flashpoint/flashpoint_connector/converter_to_stix.py", line 359, in _generate_incident_name
    "Alert: "
TypeError: can only concatenate str (not "NoneType") to str
```

The incident name is built via direct string concatenation (`"Alert: " + <field>`), and one of the alert fields used is `None` in some cases. The exception is logged by the connector's own error handler (process does not crash), but the affected alert is silently dropped and never converted into an OpenCTI incident — with no retry. This is a distinct code path / stack trace from the `media_id` issue in `_import_alerts`.

## Environment
- Connector: flashpoint (Flashpoint Alerts)
- Version: latest

## Reproducible steps
1. Deploy the flashpoint connector against a Flashpoint Alerts feed that includes at least one alert with a `null`/missing field used by `_generate_incident_name` (e.g. title/category/identifier).
2. Let the connector process alerts.
3. Observe the `TypeError: can only concatenate str (not "NoneType") to str` error raised from `_generate_incident_name`, and confirm the alert is never converted/ingested.

## Suggested fix
Guard the concatenation against `None` values in `converter_to_stix.py::_generate_incident_name` (e.g. use `str(field or "")`, or explicit null-check with a fallback value).

### LINKED ISSUE #7269: fix(flashpoint): uncaught TypeError when concatenating None media_id in _import_alerts
## Description
The Flashpoint Alerts connector raises an uncaught `TypeError` while importing alerts, when a field used in a string concatenation (related to `media_id`) is `None` instead of a string.

Observed error:
```
An error occurred while converting alert.id: <alert_id>, error: can only concatenate str (not "NoneType") to str
```

Traceback:
```
File "/opt/opencti-connector-flashpoint/flashpoint_connector/connector.py", line 395, in _import_alerts
    alert_document.get("media_id")
TypeError: can only concatenate str (not "NoneType") to str
```

This appears when the alert document's `media_id` (or a related field concatenated with it) is `None`. The exception is logged by the connector's own error handler, so the process does not crash, but the affected alert is silently dropped and never converted into an OpenCTI incident — with no retry.

## Environment
- Connector: flashpoint (Flashpoint Alerts)
- Version: latest

## Reproducible steps
1. Deploy the flashpoint connector against a Flashpoint Alerts feed that includes at least one alert document with a `null`/missing `media_id` (or related) field.
2. Let the connector process alerts.
3. Observe the `TypeError: can only concatenate str (not "NoneType") to str` error logged for the affected alert, and confirm the alert is never converted/ingested.

## Suggested fix
Guard the concatenation against `None` values (e.g. default to empty string, or skip/handle the missing field explicitly) in `connector.py::_import_alerts`, around the `media_id` handling.

### PR DESCRIPTION
**Pull request by throuxel**
### Proposed changes

* Fix an uncaught `TypeError: can only concatenate str (not "NoneType") to str` in `_generate_incident_name` by wrapping each alert field (`alert_reason`, `channel_type`, `channel_name`, `alert_id`) with `str(field or "")` before concatenation
* Fix the same class of error in `_import_alerts` when building `media_name`, by guarding both `media_id` and the guessed file extension against `None`
* Add unit tests covering incident-name generation with partial/all-None fields, and alert import with a `None` `media_id`

### Related issues

* Fixes #7268
* Fixes #7269

### Checklist

- [x] I consider the submitted work as finished
- [x] I have signed my commits using GPG key.
- [x] I tested the code for its functionality using different use cases
- [x] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### Further comments

Both issues share the same root cause: alert fields returned by the Flashpoint API can be `null`, and they were concatenated directly into strings. When any field was `None`, the connector's error handler logged the failure but silently dropped the alert, so it was never converted into an OpenCTI incident and never retried. Guarding the concatenations with a fallback empty string keeps the incident name / media name well-formed while ensuring every alert is ingested.

### CODE REVIEW

File: external-import/flashpoint/src/flashpoint_connector/converter_to_stix.py [hunk 1]
```diff
@@ -357,13 +357,13 @@ def _generate_incident_name(alert: dict) -> str:
...
+            + str(alert.get("alert_reason") or "")
             + " - "
-            + alert.get("channel_type")
+            + str(alert.get("channel_type") or "")
             + " - "
-            + alert.get("channel_name")
+            + str(alert.get("channel_name") or "")
             + " - "
-            + alert.get("alert_id")
+            + str(alert.get("alert_id") or "")
```
Comment (jabesq) [inline external-import/flashpoint/src/flashpoint_connector/converter_to_stix.py:366]: **suggestion**: Maybe add a debug log message when a field is missing, in case we have conflicting alert names.

File: external-import/flashpoint/src/flashpoint_connector/connector.py [hunk 2]
```diff
@@ -391,10 +391,9 @@ def _import_alerts(self, start_date: datetime) -> None:
                                 )
                                 processed_alert["media_content"] = media_content
                                 processed_alert["media_type"] = media_type
-                                processed_alert["media_name"] = (
-                                    alert_document.get("media_id")
-                                    + guess_file_extension
-                                )
+                                processed_alert["media_name"] = str(
+                                    alert_document.get("media_id") or ""
+                                ) + (guess_file_extension or "")
```
[the remark this answers is not in the document]
Comment (jabesq) [reply external-import/flashpoint/src/flashpoint_connector/connector.py:396]: suggestion: Maybe add a debug log message when a field is missing, or even an error when all fields are missing.

**Review by jabesq — APPROVED** [review]
